In [1]:
from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


In [2]:
%cd /content/drive/MyDrive/Colab Notebooks/project/UNET
%ls

/content/drive/MyDrive/Colab Notebooks/project/UNET
 before_TF_run3_checkpoint.pt   GPU_run_checkpoint_1.pth.tar   TF_run4_checkpoint.pt
 before_TF_run4_checkpoint.pt  'Kopi af golf_utils.py'         TF_run5_checkpoint.pt
 before_TF_run5_checkpoint.pt   metrics.csv                    TF_run6_checkpoint.pt
 before_TF_run6_checkpoint.pt   __pycache__/                   thickBowl_Run_6_GPU.ipynb
 checkpoint.pt                  run2_checkpoint.pt             thickPlate_Run_4_GPU.ipynb
 datasetHandler.py              Run_2_GPU.ipynb                thinBowl_Run_5_GPU.ipynb
 golfHandler.py                 run3_checkpoint.pt             thinPlate_Run_3_GPU.ipynb
 golf_predict_gpu.py            run4_checkpoint.pt             UNET_model_OG.ipynb
 golf_predict.py                run5_checkpoint.pt             unetSimple.py
 golf_train.py                  run6_checkpoint.pt             utils_temp.txt
 golf_utils.py                  TF_run3_checkpoint.pt


In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 1993
Number of masks: 1993
=> Loading checkpoint
Model loadet
Let's predict
Starting prediction...
Evaluating: 100% 1993/1993 [11:15<00:00,  2.95it/s]
Average Dice score: 0.9617909257552532
Average IoU score: 0.9289590279925811
Average Precision: 0.9691054377935838
Average Recall: 0.9568972998855796
Average F1 score: 0.9617909207600769
Average Accuracy: 0.9926182262920213
Average MAE: 0.007381773541171648
Average MSE: 0.007381773541171648
Average Dice Loss: 0.03820907381749668
Average BCE Loss: 0.02956431018206353
Average FPS: 66.71764645064864
Average Prediction Time per Image: 0.005159188561025601
Total Prediction Time for Epoch: 10.282262802124023
Test Dice Score: 0.9618
Test IoU Score: 0.9290
Test Precision: 0.9691
Test Recall: 0.9569
Test F1 Score: 0.9618
Test Accuracy: 0.9926
Test MAE: 0.0074
Test MSE: 0.0074
Test Dice Loss: 0.0382
Test BCE Loss: 0.0296
Test FPS: 66.72
Test

# Second prediction

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 1993
Number of masks: 1993
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 1993/1993 [01:01<00:00, 32.33it/s]
Average Dice score: 0.9617909257552532
Average IoU score: 0.9289590279925811
Average Precision: 0.9691054377935838
Average Recall: 0.9568972998855796
Average F1 score: 0.9617909207600769
Average Accuracy: 0.9926182262920213
Average MAE: 0.007381773541171648
Average MSE: 0.007381773541171648
Average Dice Loss: 0.03820907381749668
Average BCE Loss: 0.02956431018206353
Average FPS: 69.36129338389551
Average Prediction Time per Image: 0.006965971447628345
Total Prediction Time for Epoch: 13.883181095123291
Test Dice Score: 0.9618
Test IoU Score: 0.9290
Test Precision: 0.9691
Test Recall: 0.9569
Test F1 Score: 0.9618
Test Accuracy: 0.9926
Test MAE: 0.0074
Test MSE: 0.0074
Test Dice Loss: 0.0382
Test BCE Loss: 0.0296
Test FPS: 69.36
Test

# FPS FIX

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 1993
Number of masks: 1993
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 1993/1993 [01:04<00:00, 31.10it/s]
Average Dice score: 0.9617909257552532
Average IoU score: 0.9289590279925811
Average Precision: 0.9691054377935838
Average Recall: 0.9568972998855796
Average F1 score: 0.9617909207600769
Average Accuracy: 0.9926182262920213
Average MAE: 0.007381773541171648
Average MSE: 0.007381773541171648
Average Dice Loss: 0.03820907381749668
Average BCE Loss: 0.029564309801216618
Average FPS: 129.23203419181272
Average Prediction Time per Image: 0.007738019495350119
Total Prediction Time for Epoch: 15.421872854232788
Test Dice Score: 0.9618
Test IoU Score: 0.9290
Test Precision: 0.9691
Test Recall: 0.9569
Test F1 Score: 0.9618
Test Accuracy: 0.9926
Test MAE: 0.0074
Test MSE: 0.0074
Test Dice Loss: 0.0382
Test BCE Loss: 0.0296
Test FPS: 129.23
T

# TRACKMAN FPS FIX

In [8]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 112
Number of masks: 112
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 112/112 [00:15<00:00,  7.05it/s]
Average Dice score: 0.6712127297999981
Average IoU score: 0.5203112348692652
Average Precision: 0.5448235139973954
Average Recall: 0.9274700579427769
Average F1 score: 0.6712127252728458
Average Accuracy: 0.9732864858906523
Average MAE: 0.026713514141322645
Average MSE: 0.026713514141322645
Average Dice Loss: 0.32878727252994266
Average BCE Loss: 0.08330116568998035
Average FPS: 78.59115528795746
Average Prediction Time per Image: 0.012724078127316065
Total Prediction Time for Epoch: 1.4250967502593994
Test Dice Score: 0.6712
Test IoU Score: 0.5203
Test Precision: 0.5448
Test Recall: 0.9275
Test F1 Score: 0.6712
Test Accuracy: 0.9733
Test MAE: 0.0267
Test MSE: 0.0267
Test Dice Loss: 0.3288
Test BCE Loss: 0.0833
Test FPS: 78.59
Test Ave

# Testing on 112 images from other dataset of golfers

270x480

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 112
Number of masks: 112
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 112/112 [00:28<00:00,  3.90it/s]
Average Dice score: 0.6712127297999981
Average IoU score: 0.5203112348692652
Average Precision: 0.5448235139973954
Average Recall: 0.9274700579427769
Average F1 score: 0.6712127252728458
Average Accuracy: 0.9732864858906523
Average MAE: 0.026713514141322645
Average MSE: 0.026713514141322645
Average Dice Loss: 0.32878727252994266
Average BCE Loss: 0.0833011681347021
Average FPS: 19.795518611798112
Average Prediction Time per Image: 0.012646483523505074
Total Prediction Time for Epoch: 1.4164061546325684
Test Dice Score: 0.6712
Test IoU Score: 0.5203
Test Precision: 0.5448
Test Recall: 0.9275
Test F1 Score: 0.6712
Test Accuracy: 0.9733
Test MAE: 0.0267
Test MSE: 0.0267
Test Dice Loss: 0.3288
Test BCE Loss: 0.0833
Test FPS: 19.80
Test Ave

# Testing on 112 images from other dataset of golfers

160x160 test images 4

41.88

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 112
Number of masks: 112
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 112/112 [00:11<00:00,  9.33it/s]
Average Dice score: 0.7604488897167517
Average IoU score: 0.6751065394030883
Average Precision: 0.722522061009256
Average Recall: 0.82858809821879
Average F1 score: 0.7604488853653794
Average Accuracy: 0.9859099469866072
Average MAE: 0.014090052735160239
Average MSE: 0.014090052735160239
Average Dice Loss: 0.159193966537714
Average BCE Loss: 0.045547450618219694
Average FPS: 49.72433977614406
Average Prediction Time per Image: 0.010359440531049455
Total Prediction Time for Epoch: 1.160257339477539
Test Dice Score: 0.7604
Test IoU Score: 0.6751
Test Precision: 0.7225
Test Recall: 0.8286
Test F1 Score: 0.7604
Test Accuracy: 0.9859
Test MAE: 0.0141
Test MSE: 0.0141
Test Dice Loss: 0.1592
Test BCE Loss: 0.0455
Test FPS: 49.72
Test Average 

# Transfer Learning

In [ ]:
!python golf_train.py

Number of images: 70
Number of masks: 70
/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:558: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(_create_warning_msg(
Number of images: 30
Number of masks: 30
Number of images: 12
Number of masks: 12
Print loading model...
=> Loading checkpoint
Model loaded
Let's train
Epoch 1/100
  0% 0/70 [00:00<?, ?it/s]/usr/local/lib/python3.10/dist-packages/torch/autograd/graph.py:744: UserWarning: Plan failed with a cudnnException: CUDNN_BACKEND_EXECUTION_PLAN_DESCRIPTOR: cudnnFinalize Descriptor Failed cudnn_status: CUDNN_STATUS_NOT_SUPPORTED (Triggered internally at ../aten/src/ATen/native/cudnn/Conv_v8.

In [ ]:
!python golf_predict_gpu.py

Number of images: 70
Number of masks: 70
Number of images: 30
Number of masks: 30
Number of images: 12
Number of masks: 12
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 12/12 [00:15<00:00,  1.31s/it]
Average Dice score: 0.8831013921711847
Average IoU score: 0.8122990800807361
Average Precision: 0.9502473702911711
Average Recall: 0.8522145445777762
Average F1 score: 0.8831013873000622
Average Accuracy: 0.9951697530864196
Average MAE: 0.004830246902808237
Average MSE: 0.004830246902808237
Average Dice Loss: 0.11689861118793488
Average BCE Loss: 0.02036263204839391
Average FPS: 14.518709714640108
Average Prediction Time per Image: 0.023635367552439373
Total Prediction Time for Epoch: 0.28362441062927246
Test Dice Score: 0.8831
Test IoU Score: 0.8123
Test Precision: 0.9502
Test Recall: 0.8522
Test F1 Score: 0.8831
Test Accuracy: 0.9952
Test MAE: 0.0048
Test MSE: 0.0048
Test Dice Loss: 0.1169
Test BCE Loss: 0.0204
Test FPS: 14.52
Test Average Predi